azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java

```diff
@@ -322,18 +323,22 @@ private void writeObject(ObjectOutputStream out) throws IOException
 	{
 		out.defaultWriteObject();
 		
-		out.writeInt(this.bodyData.getLength());
-		out.write(this.bodyData.getArray(), this.bodyData.getArrayOffset(), this.bodyData.getLength());
+                out.writeInt(this.bodyData == null ? BODY_DATA_NULL : this.bodyData.getLength());
+                if (this.bodyData != null)
+                    out.write(this.bodyData.getArray(), this.bodyData.getArrayOffset(), this.bodyData.getLength());
 	}
 	
 	private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException
 	{
 		in.defaultReadObject();
 		
 		final int length = in.readInt();
-		final byte[] data = new byte[length];
-		in.read(data, 0, length);
-		this.bodyData = new Binary(data, 0, length);
+		if (length != BODY_DATA_NULL) {
+                    
+                    final byte[] data = new byte[length];
+                    in.readFully(data, 0, length);
```

Comment:
In general, having extra level of validation is definitely good idea.
Just to be on same page - For this specific PR - this change wouldn't benefit from that though... this desrialization is for non-amqp - serialization code paths - like, for ex: if customers want to persist EventData to a file.